### DISCUSSION TITLE
Can't read data from SerialPort, callback never called

### DISCUSSION THREAD

**Discussion by oddure** [serialport/node-serialport#2398] [Q&A] [answered]
I have a USB to serial (RS-232) cable plugged into my Windows PC and would like to read the data from the serial device.
When I tried read, the port is open but not read de data.

This is my code:

- Node Version: v16.13.1
- SerialPort Version:   9.2.8

<pre> 
const serialPort = require('serialport');

const port = new serialPort(
    'COM4',
    {
        baudRate: 9600
        , dataBits: 8,
        stopBits: 1,
        parity: 'none'
    }
)

const parser = new serialPort.parsers.Readline()

port.pipe(parser)

//abre el puerto
port.on('open', (err) => {

    //Verifica si hay error al abrir el puerto
    if (err) {
        return console.log('Error opening port: ', err.message);
    }

    console.log('Port Opened');
    //console.log(port);
});

port.on('error', function (err) {
    console.log('Error: ', err.message);
})

//lee el puerto
port.on('readable', function (err) {

    if (err) {
        return console.log('Error al leer el puerto: ', err.message);
    }

    console.log('Data:', port.read());
    port.resume();
});

parser.on('data', (line) => {
    console.log(line);
    port.resume();

});

port.on('close', function (err) {
    console.log(err);
});
</pre>

When I use the debuger, this is the result:

![Debug_Bascula_R](https://user-images.githubusercontent.com/56968638/150425015-91c3e95d-2afc-4b56-857a-f84175742d6a.png)

**Comment by imran-vz** [comment, marked as answer]
```port.resume()``` does not work if there is a readable listener.
